src/Microsoft.Health.Fhir.Shared.Core/Extensions/ReindexJobRecordExtensions.cs

```diff
@@ -26,6 +26,17 @@ public static ResourceElement ToParametersResourceElement(this ReindexJobWrapper
             parametersResource.Parameter = new List<Parameters.ParameterComponent>();
             parametersResource.Parameter.Add(new Parameters.ParameterComponent() { Name = JobRecordProperties.Id, Value = new FhirString(job.Id) });
 
+            if (job.Error != null && job.Error.Count > 0)
+            {
+                var outputMessages = new List<Parameters.ParameterComponent>();
+                foreach (var error in job.Error)
+                {
+                    outputMessages.Add(new Parameters.ParameterComponent() { Name = error.Code, Value = new FhirString(error.Diagnostics) });
```

Comment:
If this is a flat structure there is a helper method: `parametersResource.Add(error.Code, new FhirString(rerror.Diagnostics));`